src/roiextractors/segmentationextractor.py

```diff
@@ -173,14 +173,13 @@ def get_traces(self, roi_ids=None, start_frame=None, end_frame=None, name="raw")
         """
         if name not in self.get_traces_dict():
             raise ValueError(f"traces for {name} not found, enter one of {list(self.get_traces_dict().keys())}")
-        if roi_ids is None:
-            roi_idx_ = range(self.get_num_rois())
-        else:
+        if roi_ids is not None:
             all_ids = self.get_roi_ids()
-            roi_idx_ = [int(all_ids.index(i)) for i in roi_ids]
+            roi_idxs = [all_ids.index(i) for i in roi_ids]
         traces = self.get_traces_dict().get(name)
         if traces is not None and len(traces.shape) != 0:
-            return np.array([traces[start_frame:end_frame, idx] for idx in roi_idx_])
+            idxs = slice(None) if roi_ids is None else roi_idxs
+            return np.array(traces[start_frame:end_frame, :])[:, idxs]  # numpy fancy indexing is quickest
```

Comment:
Well, it's pretty subtle, but this is the reason it wasn't working. For whatever reason, the return was always an array casting of a list compression whose outer level is the ROI's.

Reply:
I mean, applying the fancy indexing for subset ROI's makes more sense to do to something you know is already a numpy array (some traces slicing might apply to `h5py.Dataset` objects) so I think that's covered here without the use of `np.stack`.

Reply:
Good point. Is the casting to `np.array` actually needed over there?

Reply:
In the event that the underlying `traces` are an `h5py.Dataset` I think it makes it clearer/more readable to cast an array for maximum safety to communicate that it is the `np.ndarray` which has the fancy indexing applied to it

Reply:
All right.

Reply:
I do think there are probably even better ways around everything in this function. Probably in a follow-up